indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java

```diff
@@ -504,10 +504,11 @@ public void run()
           SequenceMetadata sequenceToCheckpoint = null;
           for (OrderedPartitionableRecord<PartitionIdType, SequenceOffsetType> record : records) {
 
+
             // for Kafka, the end offsets are exclusive, so skip it
             if (isEndSequenceOffsetsExclusive() &&
                 createSequenceNumber(record.getSequenceNumber()).compareTo(
-                    createSequenceNumber(endOffsets.get(record.getPartitionId()))) == 0) {
+                    createSequenceNumber(endOffsets.get(record.getPartitionId()))) >= 0) {
```

Comment:
Is this possible that the sequence number of `record` is larger than the end offset? The end offset must be either `unlimited` or `max(actually consumed offsets of all replicas)`.

Reply:
i though it could be possible with transactions enabled, as some of the offsets could be used by transactional markers. I did not realize that end offset  is `unlimited` or max, where is that set ?

Reply:
The initial value of end offset is unlimited, and is updated later in `setEndOffsets()`. The endOffsets are sent from the supervisor in `setEndOffsets()` which is max of offsets of all running replicas.